megamek/src/megamek/common/Entity.java

```diff
@@ -3395,9 +3395,12 @@ public void addEquipment(Mounted mounted, int loc, boolean rearMounted)
             throws LocationFullException {
         mounted.setLocation(loc, rearMounted);
         equipmentList.add(mounted);
-        compositeTechLevel.addComponent(mounted.getType());
-        if (mounted.isArmored()) {
-            compositeTechLevel.addComponent(TA_ARMORED_COMPONENT);
+        
```

Comment:
1. These issues are unrelated and should not be in the same PR.
2. The problem here is that `compositeTechLevel` should never be null. Skipping this section of code if it has an illegal value does not fix the problem. It hides it.

Reply:
I'll implement the "right" fix for these issues in another PR then. 

For my education:

At what point is the quirks list initialized? Why does it get continuously reloaded in the middle of a tactical game?

At what point during a tactical game is the compositeTechLevel information of an on/offboard unit used?

Reply:
Entity ln822 `quirks.initialize();`
I didn't find any other areas that call initialize in Quirks.java outside of Entity and AbstractOptions